src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/SwapEndpointSpec.groovy

```diff
@@ -1373,6 +1378,89 @@ switches"() {
         !isSwitchValid && switches.each { northbound.synchronizeSwitch(it, true)}
     }
 
+    @Tidy
+    def "Able to swap endpoints for a flow with flowLoop"() {
+        setup: "Create two flows with the same src and different dst switches"
+        def tgSwitchIds = topology.getActiveTraffGens()*.switchConnected*.dpId
+        assumeTrue("Not enough traffgen switches found", tgSwitchIds.size() > 1)
+        SwitchPair flow2SwitchPair = null
+        SwitchPair flow1SwitchPair = topologyHelper.getAllNeighboringSwitchPairs().find { firstPair ->
```

Comment:
since you are strictly bind to src/dst conditions, you may also want to loop through reversed switch pairs

Reply:
fixed